csrc/host_ir/jit.cpp

```diff
@@ -0,0 +1,383 @@
+      llvm::Intrinsic::getDeclaration(mod, llvm::Intrinsic::trap);
+  builder.CreateCall(trap_func, {});
+  builder.CreateUnreachable();
+
+  builder.SetInsertPoint(then_bb);
+
+  // Create constants for type and device from params
+  llvm::Value* dtype_constant = llvm::ConstantInt::get(int32_type, static_cast<int32_t>(type));
+  llvm::Value* device_index_constant = llvm::ConstantInt::get(int64_type, params.getDeviceId());
+
+  // Get the at::native::empty_strided_cuda function pointer (registered in the JIT)
+  llvm::Function* at_empty_strided_cuda_func = mod->getFunction("at::native::empty_strided_cuda");
+  if (at_empty_strided_cuda_func == nullptr) {
+    llvm::FunctionType* at_empty_strided_cuda_func_type = llvm::FunctionType::get(
+        void_ptr_type, {int64_ptr_type, int64_type, int64_ptr_type, int64_type, int32_type, int64_type}, false);
+    at_empty_strided_cuda_func = llvm::Function::Create(
+        at_empty_strided_cuda_func_type, llvm::Function::ExternalLinkage, "at::native::empty_strided_cuda", mod);
+  }
+
+  // Call at::native::empty_strided_cuda with constants
+  llvm::Value* result =
+      builder.CreateCall(at_empty_strided_cuda_func, {sizes_arg, ndim_arg, strides_arg, strides_ndim_arg, dtype_constant, device_index_constant});
+
+  // Return the result
+  builder.CreateRet(result);
+
+  // Verify the module
+  std::string error;
+  llvm::raw_string_ostream error_stream(error);
+  NVF_ERROR(!llvm::verifyModule(*mod, &error_stream), "LLVM module verification failed: " + error);
+
+  const bool debug_print = isDebugDumpEnabled(DebugDumpOption::HostIrJit);
+  if (debug_print) {
+    // Print the LLVM IR module
+    llvm::outs() << "=== LLVM IR ===\n";
+    mod->print(llvm::outs(), nullptr);
+  }
+}
+
+void compile(const hir::HostIrContainer* container, llvm::orc::LLJIT* jit, std::unordered_map<const kir::Allocate*, allocate_fn>& allocate_funcs_, const HostIrJitParams& params) {
+  if (allocate_funcs_.size() > 0) {
+    return;
+  }
+  
+  if (container == nullptr) {
+    NVF_ERROR(false, "container is nullptr during host ir JIT compilation");
+    return;
+  }
+  FUSER_PERF_SCOPE("HostIrJit::compile");
+  auto ctx = std::make_unique<llvm::LLVMContext>();
+  auto mod = std::make_unique<llvm::Module>(
+      "host_ir_container_" +
+          std::to_string(reinterpret_cast<uintptr_t>(container)),
+      *ctx);
+  
+  // Create global variables once for the entire module
+  createGlobalVariables(mod.get(), params);
+  
+  std::unordered_map<const kir::Allocate*, std::string> allocate_func_names;
+  for (auto expr : container->topLevelExprs()) {
+    if (auto allocate = dynamic_cast<const kir::Allocate*>(expr)) {
+      // Generate a unique function name for this allocate
+      generateAllocateFunc(allocate, mod.get(), params);
+      // Store the mapping from allocate to function name
+      allocate_func_names[allocate] = "create_tensor_from_sizes_" +
+          std::to_string(reinterpret_cast<uintptr_t>(allocate));
```

Comment:
let's generate the unique name here and pass it into `generateAllocateFunc`, instead of duplicating the name function twice.

Reply:
There's also a usage of `at::native::empty_strided_cuda`